added recommended limitation text Atoms documentation.

DawDavis opened the pull request:
Alters the text of the programmers-guide.md to include the 255 byte limitation on the 'atom' Term in AtomVM. Alters the docs of the ATOM_STR macro, to also mention the limitation.

Additionally cleaned up the Macro documentation, just to be cleaner.

Resolves #1066 

These changes are made under both the "Apache 2.0" and the "GNU Lesser General
Public License 2.1 or later" license terms (dual license).

SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later

UncleGrumpy commented on doc/src/programmers-guide.md:
```diff
@@ -55,7 +55,8 @@ In addition, several features are supported specifically for integration with mi
 
 * Bignums.  Integer values are restricted to 64-bit values.
-* Bit Syntax.  While packing and unpacking of arbitrary (but less than 64-) bit values is support, packing and unpacking of integer values at the start or end of a binary, or bordering binary packing or extraction must align on 8-bit boundaries.  Arbitrary bit length binaries are not currently supported.
+* Atoms. Atoms cannot be larger than 255 bytes (255 ascii characters) are not supported.
```
This is fine, but it might be better to list Atoms first. I think those first entries were intended to be alphabetically listed, and it makes for a smaller change.

DawDavis replied:
Done - should be in the correct order now. Can't believe I didn't catch the ordering.

UncleGrumpy reviewed (approved):
This looks fine to me. I just had one suggestion about the list order.

UncleGrumpy reviewed (changes requested):
I just noticed this was base on `main`. This should be rebased on the `release-0.6` branch.

DawDavis commented:
I rebased off of the 0.6 release, set the target to 0.6 correctly and fixed the noted documentation issue in the listing order - I'm happy to make any additional changes, but I'm hoping this is up to muster. 

Also sorry for making a mess of the rebase - I think it should be squashable in merge, but if not I can reopen this PR without the cruft.

UncleGrumpy commented:
One other thing to point out, if you edit your commit message to say:

>closes #1066

Then the issue will be closed automatically after the PR is merged (or in this case, when the changes in release-0.6 are merged into main, after this PR is merged into release-0.6).

DawDavis commented:
Added `closes #1066` text to commit.

Learning new things today.

DawDavis commented on doc/src/programmers-guide.md:
```diff
@@ -54,8 +54,9 @@ In addition, several features are supported specifically for integration with mi
+* Atoms. Atoms cannot be larger than 255 bytes (255 ascii characters) are not supported.
 * Bignums.  Integer values are restricted to 64-bit values.
-* Bit Syntax.  While packing and unpacking of arbitrary (but less than 64-) bit values is support, packing and unpacking of integer values at the start or end of a binary, or bordering binary packing or extraction must align on 8-bit boundaries.  Arbitrary bit length binaries are not currently supported.
+* Bit Syntax.  While packing and unpacking of arbitrary (but less than 64-) bit values is supported, packing and unpacking of integer values at the start or end of a binary, or bordering binary packing or extraction must align on 8-bit boundaries.  Arbitrary bit length binaries are not currently supported.
```
should be good now
